### ISSUE TITLE
Multiple definitions for some DICOM tags

### ISSUE THREAD

**Issue by khelm** [incf-nidash/dicom-ontology#3]
In the DICOM standard there are tags that have different definitions depending upon context.  I'm wondering how to handle this?  The problem is that while there is a way to handle two different labels that have the same definition (`prefLabel` and `altLabel`), I haven't found any way to handle labels that have multiple definitions.  At first I was just going to choose one of the definitions that seemed most general, but it became obvious as I went through the terms that this was not going to work.

For example, `prefLabel:"Accessory Code"` (tag = (300A,00F9))
def1 = "An identifier for the Block intended to be read by a device such as a bar code reader."
def2 = "An identifier for the Compensator intended to be read by a device such as a bar code reader."
Close, but not the same thing.

One way to handle this is to have the numerical ID not be tied to the dicom tag, so instead of label `"Accessory Code"`  having the ID dicom:dicom_300A00F9, there would be two ID's and the user could just choose which one they wanted.

Anyone have other suggestions as to how to handle this or can see potential problems with this approach?

**Comment by satra** [comment]
@khelm - it looks like in many of these cases, it's a hierarchical tree structure.
`if a==thisval then b = this meaning`

however, in the specific example you pointed to, it seems the following generic definition holds in all cases: "Machine-readable identifier for this accessory", but each alternate description refers to a possible specific nature of the parent node of the tree.

fundamentally this boils down to whether we treat every dicom tag as a URI or not. if as you suggest, we go with our own IDs, we can create unique ids for each description, even though the dicom tag could be the same for all these IDs. 

if there are not a crazy amount, can we isolate these cases and clarify with david clunie whether the common description is a satisfactory one?

**Comment by khelm** [comment]
Yes, there are three types that I've identified:
1) the type that deals with different cases giving rise to different definitions: `if a==thisval then b = this`
2) the type I quoted above for `Accessory Code` in which the definition is actually different (since it refers to different hardware in that case), but has the same general meaning.  I also should have noted that there was a third version for `Accessory Code` that is more general:
"An identifier for the accessory intended to be read by a device such as a bar-code reader."
In this case the, it would make sense to choose this general definition, assuming one exists in all cases, if we are to go the route of eliminating multiple definitions.
3) a milder version of 2) in which all the definitions are saying the same thing but using different words. Ex. `Accession Number`
`"A RIS generated number that identifies the order for the Study."`
`"A departmental IS generated number that identifies the order for the Study."`
`"An identifier of the Imaging Service Request for this Requested Procedure."`

In the case of 1) we could abstract a definition that removes the if-then part and use the most general definition if one exists, but that will take a lot of work as I've found over 700 (out of ~4400) tags with multiple definitions.  

I've written some code to collect the definitions for each multiple-def tag and print them to the screen allowing me to choose one and writing that one to a new file so I end up with a clean file with no multiples.  So just choosing one of the def's would be faster.  The problem is that some of the tags only have if-then definitions and then you would have to choose one, but then go back through and remove the if-then part.  

(As an aside, I will have to do this anyway as the def's I'm using come from an XML files that had links in the definitions that weren't carried along so there are phrases like "See for more detail" where there used to be a reflink after the "See". )

On the other hand, I modified the code that writes all of this out in a turtle file - to sequentially number new ID's for each tag definition. That's done already. 

So either way, I can do it.  The problem comes down to one of usability vs effort.  If there are multiple ID's with the same tag and label, then it might be confusing to display in protege or wherever (or worse having to just display ID's rather than labels) and potentially difficult to use as you'd have to make very specific choices between definitions - and you might not know the status of the "if" part of the if-then.
